cmd/lncli/walletrpc_active.go

```diff
@@ -396,3 +401,200 @@ func labelTransaction(ctx *cli.Context) error {
 
 	return nil
 }
+
+// fundPsbtResponse is a struct that contains JSOn annotations for nice result
+// serialization.
+type fundPsbtResponse struct {
+	Psbt  string                 `json:"psbt"`
+	Locks []*walletrpc.UtxoLease `json:"locks"`
+}
+
+var fundPsbtCommand = cli.Command{
+	Name:      "fundpsbt",
+	Category:  "On-chain",
+	Usage:     "Fund a Partially Signed Bitcoin Transaction (PSBT).",
+	ArgsUsage: "psbt",
+	Description: `
+	The fundpsbt command creates a fully populated PSBT that contains enough
+	inputs to fund the outputs specified in the template PSBT.
+
+	If there are no inputs specified in the template, coin selection is
+	performed automatically. If the template does contain any inputs, it is
+	assumed that full coin selection happened externally and no additional
+	inputs are added. If the specified inputs aren't enough to fund the
+	outputs with the given fee rate, an error is returned.
+
+	After either selecting or verifying the inputs, all input UTXOs are
+	locked with an internal app ID.
+	`,
+	Flags: []cli.Flag{
+		cli.Uint64Flag{
```

Comment:
What about an ability to specify `(pkScript, value)` tuples like with `sendmany`? This addition would make it easier to use as a replacement for bitcoind when using something like a cold card, unless we intend users to still rely on HWI partially.